### PR TITLE
In StreamerInfoActions always respect vecotr's custom-allocator

### PR DESCRIPTION
**Pull request by pcanal**
See https://mattermost.web.cern.ch/root/pl/4xn9mxn6hfy1zgog8ybe915sar
and https://gitlab.cern.ch/olupton/DictProblems

### CODE REVIEW

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

File: io/io/src/TStreamerInfoActions.cxx [hunk 1]
```diff
@@ -45,6 +45,12 @@ using namespace TStreamerInfoActions;
 
 namespace TStreamerInfoActions
 {
+   bool IsDefaultVector(TVirtualCollectionProxy &proxy)
+   {
+      return ((proxy.GetCollectionType() == ROOT::kSTLvector && !(proxy.GetProperties() & TVirtualCollectionProxy::kCustomAlloc))
+              || (proxy.GetProperties() & TVirtualCollectionProxy::kIsEmulated) );
```
Comment (Axel-Naumann) [inline io/io/src/TStreamerInfoActions.cxx]: Is this more readable?
```
if (proxy.GetCollectionType() == ROOT::kSTLvector) {
   int props = proxy.GetProperties();
   if (!(props & TVirtualCollectionProxy::kCustomAlloc)) {
      return !(props & TVirtualCollectionProxy::kIsEmulated);
   }
}
return false;
```

Comment (pcanal) [reply io/io/src/TStreamerInfoActions.cxx]: humm ... maybe more readable but also incorrect/different :)

Comment (Axel-Naumann) [reply io/io/src/TStreamerInfoActions.cxx]: Yeah, I meant *after* applying C++ auto-correction, of course ;-)

Comment (pcanal) [reply io/io/src/TStreamerInfoActions.cxx]: The auto-correct makes the change not as appeal though
```
auto props = proxy.GetProperties();
if (proxy.GetCollectionType() == ROOT::kSTLvector) {
   return  !(props & TVirtualCollectionProxy::kCustomAlloc);
} else {
   return (props & TVirtualCollectionProxy::kIsEmulated);
}
```
or
```
auto props = proxy.GetProperties();
return ((proxy.GetCollectionType() == ROOT::kSTLvector && !(props & TVirtualCollectionProxy::kCustomAlloc))
        || (props & TVirtualCollectionProxy::kIsEmulated) );
```
or
```
auto props = proxy.GetProperties();
bool isVector = proxy.GetCollectionType() == ROOT::kSTLvector;
bool hasDefaultAlloc = !(props & TVirtualCollectionProxy::kCustomAlloc);
bool isEmulated = TVirtualCollectionProxy::kIsEmulated;

return isEmulated || (isVector && hasDefaultAlloc);

**Comment by Axel-Naumann** [comment]
I *really* like the last version. That's not what I read from the original code, so at least to me this would be very helpful.

**Pull request merged**